src/main/fc/fc_msp.c

```diff
@@ -1633,15 +1663,44 @@ static mspResult_e mspFcProcessInCommand(uint8_t cmdMSP, sbuf_t *src)
         break;
 #endif
 
-#ifdef USE_RTC6705
+#if defined(USE_RTC6705) || defined(VTX_COMMON)
     case MSP_SET_VTX_CONFIG:
-        ;
-        uint16_t tmp = sbufReadU16(src);
-        if  (tmp < 40)
-            masterConfig.vtx_channel = tmp;
-        if (current_vtx_channel != masterConfig.vtx_channel) {
-            current_vtx_channel = masterConfig.vtx_channel;
-            rtc6705_soft_spi_set_channel(vtx_freq[current_vtx_channel]);
+        {
+            uint16_t tmp = sbufReadU16(src);
+#if defined(USE_RTC6705)
+            if  (tmp < 40)
+                masterConfig.vtx_channel = tmp;
+            if (current_vtx_channel != masterConfig.vtx_channel) {
+                current_vtx_channel = masterConfig.vtx_channel;
+                rtc6705_soft_spi_set_channel(vtx_freq[current_vtx_channel]);
+            }
+#else
+            if (vtxCommonGetDeviceType() != VTXDEV_UNKNOWN) {
+
+                uint8_t band    = (tmp / 8) + 1;
+                uint8_t channel = (tmp % 8) + 1;
+
+                uint8_t current_band=0, current_channel=0;
+                vtxCommonGetBandChan(&current_band,&current_channel);
+                if ((current_band != band) || (current_channel != channel))
+                    vtxCommonSetBandChan(band,channel);
+
+                if (dataSize > 2) {
```

Comment:
A better way to do this is `if (sbufBytesRemaining(src))` (see https://github.com/raphaelcoeffic/betaflight/blob/227f9f5042e787e3be8dea4ff9ce0f4b03abc3ef/src/main/fc/fc_msp.c#L1515 ).